mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -638,9 +621,9 @@ LogicalResult mlir::coalesceLoops(RewriterBase &rewriter,
         emitNormalizedLoopBounds(rewriter, loop.getLoc(), lb, ub, step);
 
     rewriter.modifyOpInPlace(loop, [&]() {
-      loop.setLowerBound(newLoopParams.lowerBound);
-      loop.setUpperBound(newLoopParams.upperBound);
-      loop.setStep(newLoopParams.step);
+      loop.setLowerBound(cast<Value>(newLoopParams.lowerBound));
```

Comment:
you would need to use `getAsConstantIndexValue(rewriter, loc, newLoopParams.lowerBound)` here and below.

Reply:
Good point, used `getValueOrCreateConstantIndexOp` to avoid wrong cast if Attribute is in `newLoopParams.x`